### PR TITLE
7 post articleid comment

### PR DESCRIPTION
**Pull request by MHWatchy**
endpoint for POST .../:article_id/comments made

### CODE REVIEW

File: __tests__/db.test.js [hunk 1]
```diff
@@ -221,3 +221,111 @@ describe("/api/articles/:article_id/comments", () => {
       })
   })
 })
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: Posting a comment returns how it looks in the table in the database", () => {
```
Comment (nc-jonathan-mcguire) [inline __tests__/db.test.js]: description isnt great - maybe returns posted comment?

File: __tests__/db.test.js [hunk 2]
```diff
@@ -221,3 +221,111 @@ describe("/api/articles/:article_id/comments", () => {
...
+  })
+  test("400: Returns an error if a valid id is entered but there is no article", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "This is truly a mind-blowing experience",
+    }
+    return request(app)
+      .post("/api/articles/7002/comments")
+      .send(newComment)
+      .expect(400)
```
Comment (nc-jonathan-mcguire) [inline __tests__/db.test.js]: this would be 404 - article not found

File: __tests__/db.test.js [hunk 3]
```diff
@@ -221,3 +221,111 @@ describe("/api/articles/:article_id/comments", () => {
...
+      body: 8,
+    }
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request")
+      })
+  })
```
Comment (nc-jonathan-mcguire) [inline __tests__/db.test.js:296]: Need a test for Status 404: Not found, username does not exist
Responds with error object including status and message.

File: models/comments.models.js [hunk 4]
```diff
@@ -8,3 +9,15 @@ exports.selectCommentsFromAnArticle = (id) => {
     return data.rows
   })
 }
+
+exports.addNewComment = (inputData, id) => {
+  const { body, username } = inputData
+  const newComment = [body, username, id, 0]
+  const formattedSQL = format(
+    "INSERT INTO comments (body, author, article_id, votes) VALUES %L RETURNING *",
```
Comment (nc-jonathan-mcguire) [inline models/comments.models.js:20]: nice use of pgformat

**Review by nc-jonathan-mcguire — CHANGES_REQUESTED** [review]
Please see my comments. Just missing a test mainly:

- [x] Status 404: Not found, username does not exist
- [x] 400 should be 404 - article not found

File: __tests__/db.test.js [hunk 5]
```diff
@@ -223,7 +223,7 @@ describe("GET /api/articles/:article_id/comments", () => {
 })
 
 describe("POST /api/articles/:article_id/comments", () => {
-  test("201: Posting a comment returns how it looks in the table in the database", () => {
+  test("201: Returns the posted comment", () => {
```
Comment (nc-mick-fay) [inline __tests__/db.test.js:226]: Would you expect a newly posted comment to have a votes property of any number?

File: __tests__/db.test.js [hunk 6]
```diff
@@ -328,4 +328,17 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+      username: "John Real Kennedy",
+      body: "This is truly a mind-blowing experience",
+    }
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Username not found")
+      })
```
Comment (nc-mick-fay) [inline __tests__/db.test.js:342]: Good extra tests to account form malformed post bodies

**Review by nc-mick-fay — CHANGES_REQUESTED** [review]
Nice changes to account for missing tests

Would you expect a new comment to have a specific vote count? In your first test for this endpoint you've said votes can be any number

**Pull request merged**